src/app/components/app/index.js

```diff
@@ -37,104 +38,41 @@ const pageMap = {
     Store.removeListener('change', this.onChangeStore);
-  }
-  render() {
-    const state = this.state;
-    const appClasses = classnames('app', {
-      'app-404': state.currentPage === 'notfound'
-    });
-    const contentClasses = classnames('app-content', {
-      takeover: this.showTakeover(),
-      disabled: !!state.modal,
-      'mobile-no-scroll': state.modal === 'blogCategories' || state.modal === 'navigation' || state.modal === 'search' || this.showTakeover()
-    });
-    let content;
-    if(state.currentPage === 'notfound') {
-      content = <div className={appClasses}>
-        <Navigation pages={state.navMain} section={this.state.currentPage.split('/')[0]} page={this.state.currentPage.split('/')[1]} takeover={this.showTakeover()} />
-        <FourOhFour {...this.state} />
-      </div>;
-    } else {
-      content = (
-        <div className={appClasses}>
-          <Meta
-            title={get(state, 'page.seo.title') || ''}
-            meta={[{
-              name: "description",
-              content: get(state, 'page.seo.desc') || ''
-            }, {
-              name: "keywords",
-              content: get(state, 'page.seo.keywords') || ''
-            }, {
-              name: "og:type",
-              content: 'website'
-            }, {
-              name: "og:title",
-              content: get(state, 'page.seo.title') || ''
-            }, {
-              name: "og:description",
-              content: get(state, 'page.seo.desc') || ''
-            }, {
-              name: "og:image",
-              content: get(state, 'page.seo.image') || ''
-            }]}
-          />
-          <EntranceTransition className="nav-wrapper">
-            <Navigation pages={state.navMain} section={state.currentPage.split('/')[0]} page={state.currentPage.split('/')[1]} takeover={this.showTakeover()} />
-          </EntranceTransition>
-          <TransitionManager component="div" className={contentClasses} duration={0}>
-            <PageContainer key={state.currentPage}>
-              {this.getPage(state.currentPage)}
-              <Footer data={state.footer} studios={state.studios} />
-            </PageContainer>
-          </TransitionManager>
-          <TransitionManager component="div" className="app__modal" duration="500">
-            {this.renderModal()}
-          </TransitionManager>
-        </div>
-      );
-    }
-    return content;
-  }
+  },
+  onChangeStore(state) {
+    this.setState(state);
+  },
+  showTakeover() {
+    const { currentPage, takeover } = this.state;
+    return currentPage === 'home' && takeover && !takeover.seen;
+  },
+  getPage(pageId) {
+    return React.createElement(pageMap[pageId], this.state);
+  },
   renderModal() {
-    const state = this.state;
-    const takeover = state.takeover;
-    const modalType = state.modal;
+    const { takeover, modal: modalType } = this.state;
     let modal;
-    if(this.showTakeover()) {
+    if (this.showTakeover()) {
```

Comment:
@davidey my personal preference is for `if (...)` too, anyway found this from our CONTRIBUTING.md https://github.com/airbnb/javascript#18.3